app/main.py

```diff
@@ -1,52 +1,56 @@
-import json
-import xml.etree.ElementTree as ET
+from app.display_strategy import (
+    DisplayStrategy,
+    ConsoleDisplay,
+    ReverseDisplay,
+)
+from app.print_strategy import PrintStrategy, ReversePrint, ConsolePrint
+from app.serialize_strategy import (
+    SerializeStrategy,
+    JsonSerializer,
+    XmlSerializer,
+)
 
 
 class Book:
-    def __init__(self, title: str, content: str):
+    def __init__(self, title: str, content: str) -> None:
         self.title = title
         self.content = content
 
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+    def perform_display(self, strategy: DisplayStrategy) -> None:
+        strategy.display(self.content)
+
+    def perform_print(self, strategy: PrintStrategy) -> None:
+        strategy.print_book(self.title, self.content)
+
+    def perform_serialize(self, strategy: SerializeStrategy) -> str:
+        return strategy.serialize(self.title, self.content)
```

Comment:
I'm not sure that there is a need to create separate function for each only one call